### ISSUE TITLE
test_pydoc fails when run directly

### ISSUE THREAD

**Issue by serhiy-storchaka** [python/cpython#75211]
BPO | [31028](https://bugs.python.org/issue31028)
--- | :---
Nosy | @vstinner, @serhiy-storchaka
PRs | <li>python/cpython#2864</li><li>python/cpython#2910</li><li>python/cpython#2911</li><li>python/cpython#2043</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-07-31.09:16:15.718>
created_at = <Date 2017-07-25.12:20:58.642>
labels = ['3.7', 'type-bug', 'tests']
title = 'test_pydoc fails when run directly'
updated_at = <Date 2017-11-06.19:30:12.333>
user = 'https://github.com/serhiy-storchaka'
```

bugs.python.org fields:
```python
activity = <Date 2017-11-06.19:30:12.333>
actor = 'nascheme'
assignee = 'none'
closed = True
closed_date = <Date 2017-07-31.09:16:15.718>
closer = 'serhiy.storchaka'
components = ['Tests']
creation = <Date 2017-07-25.12:20:58.642>
creator = 'serhiy.storchaka'
dependencies = []
files = []
hgrepos = []
issue_num = 31028
keywords = []
message_count = 6.0
messages = ['299059', '299295', '299320', '299321', '299322', '299546']
nosy_count = 2.0
nosy_names = ['vstinner', 'serhiy.storchaka']
pr_nums = ['2864', '2910', '2911', '2043']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue31028'
versions = ['Python 2.7', 'Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by serhiy-storchaka** [comment]
```sh
$ ./python Lib/test/test_pydoc.py
...
```

======================================================================
ERROR: test_mixed_case_module_names_are_lower_cased (__main__.PydocDocTest)
\----------------------------------------------------------------------
```pytb
Traceback (most recent call last):
  File "Lib/test/test_pydoc.py", line 462, in test_mixed_case_module_names_are_lower_cased
    self.assertIn('xml.etree.elementtree', doc_link)
  File "/home/serhiy/py/cpython/Lib/unittest/case.py", line 1102, in assertIn
    if member not in container:
TypeError: argument of type 'NoneType' is not iterable
```


----------------------------------------------------------------------
Ran 49 tests in 5.716s

```pytb
FAILED (errors=1, skipped=3)
Traceback (most recent call last):
  File "Lib/test/test_pydoc.py", line 1106, in <module>
    test_main()
  File "/home/serhiy/py/cpython/Lib/test/support/__init__.py", line 2071, in decorator
    return func(*args)
  File "Lib/test/test_pydoc.py", line 1100, in test_main
    PydocWithMetaClasses,
  File "/home/serhiy/py/cpython/Lib/test/support/__init__.py", line 1940, in run_unittest
    _run_suite(suite)
  File "/home/serhiy/py/cpython/Lib/test/support/__init__.py", line 1905, in _run_suite
    raise TestFailed(err)
test.support.TestFailed: Traceback (most recent call last):
  File "Lib/test/test_pydoc.py", line 462, in test_mixed_case_module_names_are_lower_cased
    self.assertIn('xml.etree.elementtree', doc_link)
  File "/home/serhiy/py/cpython/Lib/unittest/case.py", line 1102, in assertIn
    if member not in container:
TypeError: argument of type 'NoneType' is not iterable
```

**Comment by vstinner** [comment]
New changeset fd46561167af6cd697191dd7ebb8c2fef5ad6493 by Victor Stinner in branch 'master':
bpo-31028: Fix test_pydoc when run directly (bpo-2864)
https://github.com/python/cpython/commit/fd46561167af6cd697191dd7ebb8c2fef5ad6493

**Comment by vstinner** [comment]
New changeset fd6736d17901e539d6a9ddf50a9927a3a97a9c06 by Victor Stinner in branch '2.7':
bpo-31028: Fix test_pydoc when run directly (bpo-2864) (bpo-2911)
https://github.com/python/cpython/commit/fd6736d17901e539d6a9ddf50a9927a3a97a9c06

**Comment by vstinner** [comment]
New changeset d019c7965d411b2132f68ed6bb33e538b4d9eba3 by Victor Stinner in branch '3.6':
bpo-31028: Fix test_pydoc when run directly (bpo-2864) (bpo-2910)
https://github.com/python/cpython/commit/d019c7965d411b2132f68ed6bb33e538b4d9eba3

**Comment by vstinner** [comment]
Bug fixed in Python 2.7, 3.6 and master.

**Comment by serhiy-storchaka** [comment]
Thanks Victor. But please don't forget to edit commit messages before merging. Commit messages of above commits look confusing.
